aws-sdk-core/lib/aws-sdk-core/shared_credentials.rb

```diff
@@ -67,16 +67,30 @@ def default_path
 
     def load_from_path
       profile = load_profile
-      @credentials = Credentials.new(
+      credentials = Credentials.new(
         profile['aws_access_key_id'],
         profile['aws_secret_access_key'],
         profile['aws_session_token']
       )
+      @credentials = if role_arn = profile['role_arn']
+        AssumeRoleCredentials.new(
+          role_session_name: [*('A'..'Z')].sample(16).join,
```

Comment:
Actually you propose the cosmetic change. :) If this one is only blocker I'll do it ;)